pysteps/blending/steps.py

```diff
@@ -694,15 +694,11 @@ def forecast(
         # 5. Repeat precip_cascade for n ensemble members
         # First, discard all except the p-1 last cascades because they are not needed
         # for the AR(p) model
-        precip_cascade = [
-            precip_cascade[i][-ar_order:] for i in range(n_cascade_levels)
-        ]
-
-        precip_cascade = [
-            [precip_cascade[j].copy() for j in range(n_cascade_levels)]
-            for i in range(n_ens_members)
-        ]
-        precip_cascade = np.stack(precip_cascade)
+
+        precip_cascade = np.stack(
```

Comment:
In this new version, you use a repeated reference to the same sliced data, which is more memory efficient but can lead to unintended side effects if the data is modified later. The previous version used deep copies to ensure that modifications to one ensemble member do not affect the others. Did you consider such aspects?